src/coreclr/vm/stubhelpers.cpp

```diff
@@ -476,17 +476,37 @@ FCIMPL1(void*, StubHelpers::GetDelegateTarget, DelegateObject *pThisUNSAFE)
 }
 FCIMPLEND
 
+#include <optsmallperfcritical.h>
+FCIMPL2(FC_BOOL_RET, StubHelpers::TryGetStringTrailByte, StringObject* thisRefUNSAFE, UINT8 *pbData)
+{
+    FCALL_CONTRACT;
 
+    STRINGREF thisRef = ObjectToSTRINGREF(thisRefUNSAFE);
+    FC_RETURN_BOOL(thisRef->GetTrailByte(pbData));
+}
+FCIMPLEND
+#include <optdefault.h>
 
-FCIMPL2(void, StubHelpers::ThrowInteropParamException, UINT resID, UINT paramIdx)
+extern "C" void QCALLTYPE StubHelpers_SetStringTrailByte(QCall::StringHandleOnStack str, UINT8 bData)
```

Comment:
Is it worth doing this QCall as a `SuppressGCTransition` QCall?

Reply:
This cannot be SuppressGCTransition QCall. It can allocate memory and throw exceptions.

This QCall handles a corner-case for VB6 interop where BSTRs can be used to pass around arrays of bytes. It is not a perf sensitive QCall.

(If it was a perf sensitive QCall, it can be split into SuppressGCTransition fast path and full transition slow path; or the fast path can be rewritten in C# as David pointed out. None of it is worth it at this point.)